api/include/opentelemetry/metrics/meter.h

```diff
@@ -36,7 +36,7 @@ class Meter
    * @return a shared pointer to the created Counter.
    */
 
-  virtual nostd::shared_ptr<Counter<long>> CreateLongCounter(
+  virtual nostd::shared_ptr<Counter<uint64_t>> CreateUInt64Counter(
```

Comment:
Could this increase the cost when compiled as 32-bit (like for IoT)?

Reply:
yes, it would. We can plan to add uint32_t support in future if there are IoT requirements. That would be ABI breaking change, but should be fine, I hope.